packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -190,17 +191,15 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
 
         this.toDispose.push(this.terminalWatcher.onTerminalError(({ terminalId, error }) => {
             if (terminalId === this.terminalId) {
+                this.setExitStatus();
                 this.dispose();
-                this.onTermDidClose.fire(this);
-                this.onTermDidClose.dispose();
                 this.logger.error(`The terminal process terminated. Cause: ${error}`);
             }
         }));
-        this.toDispose.push(this.terminalWatcher.onTerminalExit(({ terminalId }) => {
+        this.toDispose.push(this.terminalWatcher.onTerminalExit(({ terminalId, code, signal }) => {
```

Comment:
Remove `signal` you are not using it.

Reply:
Resolved in commit : [Fixed exitStatus undefined when closing terminal manually](https://github.com/eclipse-theia/theia/pull/11175/commits/962e252d55d4608f218968bd2be5e7dfcae304f4)